src/interface/variable_pack.hpp

```diff
@@ -573,17 +579,20 @@ void FillSwarmVarView(const vpack_types::SwarmVarList<T> &vars, PackIndexMap *vm
   int vindex = 0;
   int sparse_start;
   std::string sparse_name;
-  // TODO(BRR) Remove the logic for sparse variables
   for (const auto v : vars) {
-    if (vmap != nullptr) {
-      vmap->insert(sparse_name, IndexPair(sparse_start, vindex - 1));
-      sparse_name = "";
-    }
     int vstart = vindex;
     // Reusing ViewOfParArrays which expects 3D slices
     host_view(vindex++) = v->data.Get(0, 0, 0);
+
+    std::vector<int> shape;
+    if (v->GetDim(2) > 1) shape.push_back(v->GetDim(2));
+    if (v->GetDim(3) > 1) shape.push_back(v->GetDim(3));
+    if (v->GetDim(4) > 1) shape.push_back(v->GetDim(4));
+    if (v->GetDim(5) > 1) shape.push_back(v->GetDim(5));
+    if (v->GetDim(6) > 1) shape.push_back(v->GetDim(6));
```

Comment:
Why is the shape flexible here? As far as I can in other places it's hardocded to 6 with unused dims = 1.

Reply:
Is this not consistent with e.g. `FillVarView` (line ~550)? Maybe I'm misunderstanding

Reply:
You just updated the corresponding lines with PR #678 last week and I'm probably more confused now.
Shape is not necessarily tied to the `dims_` of a variable, isn't it?
Following https://github.com/lanl/parthenon/blob/develop/docs/interface/containers.md#meshblockdata a cell centered vector variable could have `shape = std::vector<int>({3});` but the dims of the actual variable would be different (with the first three being tied to the mesh and only the forth one tied to the vector component).

Reply:
Ah yes this needs to be updated for consistency with #678 (done). Yes, `shape` is not tied to the `dims_` of a variable, but the relationship between `shape` and `dims_` should be fixed based on what kind of variable it is. For a cell-centered variable, the first three indices are always the `i,j,k` spatial indices, leaving three more indices for `shape`, whereas for particle variable the first index is always the swarm index, leaving five more indices for `shape`. This is I think correctly captured by the different functions `FillSwarmVarView` and `FillVarView`. Maybe I am missing something though -- do you see a case that the current logic in `variable_pack.hpp` will fail for?